### PR TITLE
Implement viewCompletedTasks method of TaskManager class

### PR DESCRIPTION
**Pull request by Mustapha7018**
This pull request introduces the **viewCompletedTasks** method to the TaskManager class in our command-line task management application. The method allows users to view a list of all tasks that have been marked as completed.

**Key Features:**

- **Filtering Completed Tasks:** The method filters the task list to include only those tasks with a status of **Status.completed.**
- **Edge Case Handling:** The method handles edge cases, such as when there are no completed tasks, by providing appropriate feedback to the user.

### CODE REVIEW

File: task-manager.dart [hunk 1]
```diff
@@ -45,14 +45,27 @@ class Task {
...
   // Add task method
   void addTask() {}
 
   // View all tasks
   void viewAllTask() {}
 
   // View all completed tasks
-  void viewCompletedTasks() {}
+  void viewCompletedTasks() {
+    List<Task> completedTasks = _tasks.where((task) => task.getStatus() == Status.completed).toList();
```
Comment (BambinoRiya) [inline task-manager.dart:59]: Good use of the where method to filter tasks based on status.

File: task-manager.dart [hunk 2]
```diff
@@ -45,14 +45,27 @@ class Task {
...
+
+    if (completedTasks.isEmpty) {
```
Comment (BambinoRiya) [inline task-manager.dart:61]: nit consider adding a blank line before the if statement for better readability.

File: task-manager.dart [hunk 3]
```diff
@@ -45,14 +45,27 @@ class Task {
 }
 
 class TaskManager {
+
+  List<Task> _tasks = [];
```
Comment (BambinoRiya) [inline task-manager.dart:49]: FYI:
You might want to add error handling in case the _tasks list is null or empty.

**Review by BambinoRiya — CHANGES_REQUESTED** [review]
FYI You might want to add error handling in case the _tasks list is null or empty.

File: task-manager.dart (hunk 2 above)
Comment (Mustapha7018) [reply task-manager.dart:61]: I already did that :)...

**Pull request merged**